File: client/python/mesh/prsim/graphics.py
#
# Copyright (c) 2000, 2023, trustbe and/or its affiliates. All rights reserved.
# TRUSTBE PROPRIETARY/CONFIDENTIAL. Use is subject to license terms.
#
#

from abc import ABC, abstractmethod
from typing import Dict

from mesh.kinds.captcha import Captcha
from mesh.macro import spi, mpi


@spi("mesh")
class Graphics(ABC):

    @abstractmethod
    @mpi("mesh.graphics.captcha.apply")
    def captcha(self, kind: str, features: Dict[str, str]) -> Captcha:
        """
        Apply a graphics captcha.
        :param kind:
        :param features:
        :return:
        """
        pass

    @abstractmethod
    @mpi("mesh.graphics.captcha.verify")
    def verify(self, mno: str, value: str) -> bool:
        """
        Verify a graphics captcha value.
        :param mno:
        :param value:
        :return:
        """
        pass

File: client/java/src/main/java/io/be/mesh/prsim/Graphics.java
/*
 * Copyright (c) 2000, 2023, trustbe and/or its affiliates. All rights reserved.
 * TRUSTBE PROPRIETARY/CONFIDENTIAL. Use is subject to license terms.
 *
 *
 */
package io.be.mesh.prsim;

import io.be.mesh.macro.MPI;
import io.be.mesh.macro.SPI;
import io.be.mesh.struct.Captcha;

import java.util.Map;

/**
 * @author coyzeng@gmail.com
 */
@SPI("mesh")
public interface Graphics {

    /**
     * Apply a graphics captcha.
     */
    @MPI("mesh.graphics.captcha.apply")
    Captcha captcha(String kind, Map<String, String> features);

    /**
     * Verify a graphics captcha value.
     */
    @MPI("mesh.graphics.captcha.verify")
    boolean verify(String mno, String value);
}
